### ISSUE TITLE
@@AUTOID@@@ not consistant

### ISSUE THREAD

**Issue by mattgig** [justinhunt/moodle-filter_generico#44]
When you add an @@AUTOID@@ to the "body of template X" section and again to the "End tags (template X") section in Moodle 4.2, two different strings are generated.

<img width="612" alt="image" src="https://github.com/justinhunt/moodle-filter_generico/assets/7272764/d404cf06-4be0-4a23-baeb-34ab78d46d6e">

The above template results in two different AUTOIDs:

<img width="199" alt="image" src="https://github.com/justinhunt/moodle-filter_generico/assets/7272764/67c66ab0-667b-4508-85fe-4c5a5c887c8a">

This is contrary to what you would expect, namely the AUTOID to be consistant within the filter area. And it makes selecting corresponding DOM elements very difficult.

Is this the intended behaviour or a bug?

**Comment by justinhunt** [comment]
Yes you are correct that it is inconsistent. It should not be this way, but I have not yet figured out how to implement it. This is because the start and close tags are treated as two separate filter strings, witjh two separate passes of the filter processing script. 

In practice I just do all the processing that requires the @@AUTOID@@ in the start tag or js, and leave only static html in the end tag. But sometimes that really does make it hard.
